src/types/path-set.ts

```diff
@@ -1,120 +1,248 @@
-  {
-    inherits: Array,
-    statics: {
-      from(value) {
-        return ensureArrayLikeIs(Path, value).withChildren(Hop);
-      },
-    },
-    toJSON() {
-      return this.map((k) => k.toJSON());
-    },
-  },
-  undefined
-);
-
-const PathSet = makeClass(
-  {
-    mixins: SerializedType,
-    inherits: Array,
-    statics: {
-      from(value) {
-        return ensureArrayLikeIs(PathSet, value).withChildren(Path);
-      },
-      fromParser(parser) {
-        const pathSet = new this();
-        let path;
-        while (!parser.end()) {
-          const type = parser.readUInt8();
-          if (type === PATHSET_END_BYTE) {
-            break;
-          }
-          if (type === PATH_SEPARATOR_BYTE) {
-            path = null;
-            continue;
-          }
-          if (!path) {
-            path = new Path();
-            pathSet.push(path);
-          }
-          path.push(Hop.parse(parser, type));
-        }
-        return pathSet;
-      },
-    },
-    toJSON() {
-      return this.map((k) => k.toJSON());
-    },
-    toBytesSink(sink) {
-      let n = 0;
-      this.forEach((path) => {
-        if (n++ !== 0) {
-          sink.put([PATH_SEPARATOR_BYTE]);
-        }
-        path.forEach((hop) => {
-          sink.put([hop.type()]);
-          hop.account && hop.account.toBytesSink(sink);
-          hop.currency && hop.currency.toBytesSink(sink);
-          hop.issuer && hop.issuer.toBytesSink(sink);
-        });
-      });
-      sink.put([PATHSET_END_BYTE]);
-    },
-  },
-  undefined
-);
+/**
+ * The object representation of a Hop, an issuer AccountID, an account AccountID, and a Currency
+ */
+interface HopObject {
+  issuer?: string;
+  account?: string;
+  currency?: string;
+}
+
+/**
+ * Serialize and Deserialize a Hop
+ */
+class Hop extends SerializedTypeClass {
+  constructor(bytes: Buffer) {
+    super(bytes);
+  }
+
+  /**
+   * Create a Hop from a HopObject
+   * 
+   * @param value Either a hop or HopObject to create a hop with
+   * @returns a Hop
+   */
+  static from(value: Hop | HopObject): Hop {
+    if (value instanceof this) {
+      return value;
+    }
+
+    const bytes: Array<Buffer> = [Buffer.from([0])];
+
+    if (value.account) {
+      bytes.push(AccountID.from(value.account).toBytes());
+      bytes[0][0] |= TYPE_ACCOUNT;
+    }
+
+    if (value.currency) {
+      bytes.push(Currency.from(value.currency).toBytes());
+      bytes[0][0] |= TYPE_CURRENCY;
+    }
+
+    if (value.issuer) {
+      bytes.push(AccountID.from(value.issuer).toBytes());
+      bytes[0][0] |= TYPE_ISSUER;
+    }
+
+    return new Hop(Buffer.concat(bytes));
+  }
+
+  /**
+   * Construct a Hop from a BinaryParser
+   * 
+   * @param parser BinaryParser to read the Hop from
+   * @returns a Hop
+   */
+  static fromParser(parser: BinaryParser): Hop {
+    const type = parser.readUInt8();
+    const bytes: Array<Buffer> = [Buffer.from([type])];
+
+    type & TYPE_ACCOUNT && bytes.push(parser.read(AccountID.width));
```

Comment:
optional: 
I know you're just translating code, but I generally think it's more readable to write this as:
```js
if (type & TYPE_ACCOUNT) {
  bytes.push(parser.read(AccountID.width));
}
```

I prefer brackets but you could also do:
```js
if (type & TYPE_ACCOUNT) bytes.push(parser.read(AccountID.width));
```

if you wanted to be brief. 

Both of these strike me as more idiomatic and readable than the existing code, unless there's something magical going on that I don't understand.

Reply:
This is also true on :92 so if you address this please be consistent.